### ISSUE TITLE
Couldn't start astro server with Bun v1.1.1+ca1dbb4eb on windows

### ISSUE THREAD

**Issue by RayyanNafees** [oven-sh/bun#9980]
The setup was smooth (though library installation step wasnt faster than pnpm)

But when trying to run the server as per the [docs](https://bun.sh/guides/ecosystem/astro)

Encountered the following error :
```powershell
44331 |         sysPath$2.resolve(path, evPath), KEY_LISTENERS, sysPath$2.join(path, evPath)
44332 |       );
44333 |     }
44334 |   };
44335 |   try {
44336 |     return fs$7.watch(path, options, handleEvent);
                   ^
EBUSY: EBUSY watching "D:/Astro/bunastro"
 syscall: "watch"
   path: "D:/Astro/bunastro"

      at new FSWatcher (node:fs:105:27)
      at watch2 (node:fs:284:199)
      at createFsWatchInstance (D:\Astro\bunastro\node_modules\vite\dist\node\chunks\dep-whKeNLxG.js:44336:12)
      at setFsWatchListener (D:\Astro\bunastro\node_modules\vite\dist\node\chunks\dep-whKeNLxG.js:44383:15)
      at _watchWithNodeFs (D:\Astro\bunastro\node_modules\vite\dist\node\chunks\dep-whKeNLxG.js:44538:14)
      at D:\Astro\bunastro\node_modules\vite\dist\node\chunks\dep-whKeNLxG.js:44774:14
```

Using Bun v1.1.0 was running the server but as soon as any request was sent to the localhost the server used to crash

### Error:
![image](https://github.com/oven-sh/bun/assets/59685062/fa3faebd-77d0-4043-b215-14e00a633343)

### Windows Information:
![image](https://github.com/oven-sh/bun/assets/59685062/8b14d08c-0f37-42ea-9a9e-82518b6f026e)

**Comment by emrecancorapci** [comment]
Getting the exact same error with Vite v5.2.8.

**Comment by Jarred-Sumner** [comment]
This will be fixed in Bun v1.1.2, which releases today. 

In about an hour, feel free to run `bun upgrade --canary` and give the fix a try
